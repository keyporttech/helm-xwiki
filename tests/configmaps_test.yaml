suite: test configmaps
templates:
  - configmaps.yaml
tests:
  - it: should set values properly
    release:
      name: release
      service: tiller
    asserts:
      - isKind:
          of: ConfigMap
      - isAPIVersion:
          of: v1
      - equal:
          path: metadata.name
          value: release-xwiki
      - equal:
          path: metadata.labels.app
          value: release-xwiki
      - matchRegex:
          path: metadata.labels.chart
          value: xwiki-*
      - equal:
          path: metadata.labels.release
          value: release
      - equal:
          path: metadata.labels.heritage
          value: Tiller
      - matchSnapshot:
          path: spec
  - it: should set mysql config values
    set:
      mysql.mysqlUser: xwiki
      mysql.mysqlDatabase: xwikidb
      mysql.enabled: true
      postgresql.enabled: true
    release:
      name: release
    asserts:
      - equal:
          path: data.DB_USER
          value: xwiki
      - equal:
          path: data.DB_HOST
          value: release-mysql
      - equal:
          path: data.DB_DATABASE
          value: xwikidb
  - it: should set postgres config values
    set:
      postgresql.postgresqlUsername: xwiki
      postgresql.postgresqlDatabase: xwikidb
      mysql.enabled: false
      postgresql.enabled: true
    release:
      name: release
    asserts:
      - equal:
          path: data.DB_USER
          value: xwiki
      - equal:
          path: data.DB_HOST
          value: release-postgresql
      - equal:
          path: data.DB_DATABASE
          value: xwikidb
  - it: should set externalDB config values
    set:
      externalDB.user: xwiki
      externalDB.host: xwiki-host
      externalDB.database: xwikidb
      mysql.enabled: false
      postgresql.enabled: false
    asserts:
      - equal:
          path: data.DB_USER
          value: xwiki
      - equal:
          path: data.DB_HOST
          value: xwiki-host
      - equal:
          path: data.DB_DATABASE
          value: xwikidb
