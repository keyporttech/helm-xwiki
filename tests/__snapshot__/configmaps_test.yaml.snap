should set values properly:
  1: |
    null
